react/features/stream-effects/virtual-background/JitsiStreamBackgroundEffect.js

```diff
@@ -114,13 +121,26 @@ export default class JitsiStreamBackgroundEffect {
             this._inputVideoElement.width,
             this._inputVideoElement.height
         );
+        if (this._options.virtualBackground.backgroundType === VIRTUAL_BACKGROUND_TYPE.DESKTOP_SHARE) {
+            this._outputCanvasCtx.restore();
+        }
         this._outputCanvasCtx.globalCompositeOperation = 'source-in';
         this._outputCanvasCtx.filter = 'none';
 
         // Draw the foreground video.
         //
+        if (this._options.virtualBackground.backgroundType === VIRTUAL_BACKGROUND_TYPE.DESKTOP_SHARE) {
+            // save current context before applying transformations
+            this._outputCanvasCtx.save();
 
+            // flip the canvas and prevent mirror behaviour
+            this._outputCanvasCtx.scale(-1, 1);
+            this._outputCanvasCtx.translate(-this._outputCanvasElement.width, 0);
+        }
         this._outputCanvasCtx.drawImage(this._inputVideoElement, 0, 0);
+        if (this._options.virtualBackground.backgroundType === VIRTUAL_BACKGROUND_TYPE.DESKTOP_SHARE) {
```

Comment:
Instead of adding so many if conditions and calling `ctx.drawImage` multiple times, can you please initialize all the vars based on the background type and call it only once. The code is hard to read the way it is right now. Also please get rid of extra new lines and fix the comments ? They should start with an uppercase letter and end with a period.